src/core/contracts/identity.test.ts

```diff
@@ -331,74 +339,373 @@ describe("identity", () => {
+
+      it("returns empty list of identities", async () => {
+        const fakeAddress = "0x1ea32de10d5a18e55debaf379b26cc0c6952b168";
+
+        const result = await getDelegeteIdentitiesFor(fakeAddress);
+
+        expect(result).toEqual([]);
       });
     });
 
-    describe("when logs have been emmited for delegate", () => {
+    describe("when a public address is associated (belongs) to a single identity contract", () => {
+      let contractOwner: EthereumAddress;
+      let contractAddress: EthereumAddress;
+      let signer: ethers.Signer;
+
       beforeAll(async () => {
+        await snapshotHardhat(provider);
+      });
+
+      beforeAll(async () => {
+        signer = getSignerForAccount(10);
         contractOwner = await signer.getAddress();
+
         const identityContract = await new Identity__factory(signer).deploy(contractOwner);
         await identityContract.deployed();
         contractAddress = identityContract.address;
+      });
+
+      afterAll(async () => {
+        await revertHardhat(provider);
+      });
 
-        await upsertDelegate(contractAddress, NON_OWNER, 0x1);
-        blockNumberForRemoval = (await provider.getBlockNumber()) + 5;
-        await removeDelegate(contractAddress, NON_OWNER, blockNumberForRemoval);
+      it("returns one identity contract address associated with public address", async () => {
+        const result = await getDelegeteIdentitiesFor(contractOwner);
+
+        expect(result.length).toBe(1);
+        expect(result).toEqual([contractAddress]);
       });
 
-      it("returns logs for delegate removal", async () => {
-        const expected = {
-          endBlock: blockNumberForRemoval,
-          delegate: ethers.utils.getAddress(NON_OWNER),
+      describe("and an update is made to change permission", () => {
+        beforeEach(async () => {
+          await upsertDelegate(contractAddress, contractOwner, 0x1, { signer: signer });
+        });
+
+        it("returns one identity contract address associated with public address", async () => {
+          const result = await getDelegeteIdentitiesFor(contractOwner);
+
+          expect(result.length).toBe(1);
+          expect(result).toEqual([contractAddress]);
+        });
+      });
+
+      describe("and delegate is removed immediately", () => {
+        beforeAll(async () => {
+          await snapshotHardhat(provider);
+        });
+
+        afterAll(async () => {
+          await revertHardhat(provider);
+        });
+
+        beforeAll(async () => {
+          await removeDelegate(contractAddress, contractOwner, 0x1, { signer: signer });
+        });
+
+        it("returns zero identity contract address associated with public address", async () => {
+          const result = await getDelegeteIdentitiesFor(contractOwner);
+
+          expect(result.length).toBe(0);
+          expect(result).toEqual([]);
+        });
+      });
+
+      describe("and delegate is scheduled to be removed after 10 blocks", () => {
```

Comment:
should we show a use case here of the delegate being removed after block 10?

Reply:
I think if they are removed right now or in 10 blocks in the same thing right?